app/models/carrier.rb

```diff
@@ -0,0 +1,13 @@
+class Carrier < ActiveRecord::Base
+  validates :request, presence: true
+  validates :response, presence: true
+
```

Comment:
You could make this one line by doing `validates_presence_of :request, :response` if you wanted. Just fyi.